sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -846,9 +847,9 @@ class Analyzer(
    */
   object ResolveTempViews extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
-      case u @ UnresolvedRelation(ident, _) =>
+      case u @ UnresolvedRelation(ident, _, _) =>
         lookupTempView(ident).getOrElse(u)
```

Comment:
shall we fail if the temp view is not a streaming plan but the `isStreaming` flag is true?

Reply:
Probably vice versa.

Reply:
I'm not sure about the other way around. We have `SparkSession.table`, which can read both batch or streaming temp views. We shouldn't break it.

It's a bit weird if `DataFramaReader.table` can read streaming temp view, but this is the existing behavior and probably is fine.

Reply:
Just curious, is it end user's responsibility to know about whether the temp view is from batch or streaming, so that they can correctly call write or writeStream? Without thinking of SparkSession.table I assume it's clear as end user will match the reader side and writer side clearly (read/write or readStream/writeStream), and it looks a bit confusing.

If `DataFrameReader.table` allows streaming temp view, then I guess read/writeStream pair is possible which is a bit confusing.

Reply:
I think it's less confusing if `DataFrameReader.table` fails on reading streaming temp view while `SparkSession.table` works. But it's a breaking change and we at least shouldn't do it in this PR.

Reply:
OK I agree it should be better to fix in different PR. I still think we shouldn't support confused things just because we supported them. The fix would depend on change of the PR (isStreaming flag) - I'll wait for this PR and try to fix it after the PR.